appinventor/appengine/src/com/google/appinventor/client/style/GSoC/UIFactoryGSoC.java

```diff
@@ -0,0 +1,42 @@
+package com.google.appinventor.client.style.GSoC;
+
+import com.google.appinventor.client.UIStyleFactory;
+import com.google.appinventor.client.editor.simple.SimpleNonVisibleComponentsPanel;
+import com.google.appinventor.client.editor.simple.SimpleVisibleComponentsPanel;
+import com.google.appinventor.client.editor.youngandroid.YaFormEditor;
+import com.google.appinventor.client.explorer.folder.ProjectFolder;
+import com.google.appinventor.client.explorer.youngandroid.ProjectList;
+import com.google.appinventor.client.wizards.youngandroid.NewYoungAndroidProjectWizard;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.uibinder.client.UiBinder;
+import com.google.gwt.uibinder.client.UiTemplate;
+
+public class UIFactoryGSoC extends UIStyleFactory {
+
+  @Override
+  public ProjectList createProjectList() {
+    return new ProjectListGSoC();
+  }
+
+  @Override
+  public ProjectFolder createProjectFolder(String name, long dateCreated, long dateModified, ProjectFolder parent) {
+    return new ProjectFolderGSoC(name, dateCreated, dateModified, parent);
+  }
+
+  @Override
+  public ProjectFolder createProjectFolder(String name, long dateCreated, ProjectFolder parent) {
+    return new ProjectFolderGSoC(name, dateCreated, parent);
+  }
+
+  @Override
+  public ProjectFolder createProjectFolder(JSONObject json, ProjectFolder parent) {
+    return new ProjectFolderGSoC(json, parent);
+  }
+
+  @Override
+  public SimpleVisibleComponentsPanel createSimpleVisibleComponentsPanel
+      (YaFormEditor editor, SimpleNonVisibleComponentsPanel nonVisPanel) {
+    return new SimpleVisibleComponentsPanelGSoC(editor, nonVisPanel);
+  }}
```

Comment:
Code style: The closing curly brace for the class should be on its own line. Also, there's more whitespace at the end of the file here than strictly necessary.